packages/shared/lib/auxiliary/popup/stores/popup-state.store.ts

```diff
@@ -7,5 +7,5 @@ import { PopupProps } from '../types'
 export const popupState = writable<IPopupState>(DEFAULT_POPUP_STATE)
 
 export function updatePopupProps(payload: PopupProps): void {
-    popupState?.update((state) => ({ ...state, ...payload }))
+    popupState?.update((state) => ({ ...state, props: { ...state?.props, ...payload } }))
```

Comment:
If the props are previously undefined, will this throw a type error?

Reply:
Nope this works because of the spread operator on the optionally chained state?.props inside an object